lib/ansible/modules/cloud/amazon/kinesis_stream.py

```diff
@@ -804,6 +858,36 @@ def update(client, current_stream, stream_name, retention_period=None,
             )
             return success, changed, err_msg
 
+    if current_stream['OpenShardsCount'] != number_of_shards:
+        success, err_msg = (
+            update_shard_count(client, stream_name, number_of_shards, check_mode=check_mode)
+        )
+
+        if not success:
```

Comment:
I'm not sure about this. Will tags get correctly updated if the number of shards isn't changing?